crates/sncast/src/lib.rs

```diff
@@ -168,11 +168,12 @@ pub async fn get_account<'a>(
     keystore: Option<Utf8PathBuf>,
 ) -> Result<SingleOwnerAccount<&'a JsonRpcClient<HttpTransport>, LocalWallet>> {
     let chain_id = get_chain_id(provider).await?;
-    let account = if let Some(keystore) = keystore {
+    let mut account = if let Some(keystore) = keystore {
         get_account_from_keystore(provider, chain_id, &keystore, account)?
     } else {
         get_account_from_accounts_file(account, accounts_file, provider, chain_id)?
     };
+    let account = account.set_block_id(get_block_id("pending")?).clone();
```

Comment:
Can you clarify the reasoning behind setting the block id here, rather than using a default value for `"pending"` within the `get_nonce` function?

Reply:
We want to use a pending block as default. There is also an issue that will add a flag that allows one to choose which block will be used. IMO is a better way of using starknet-rs which is under sncast. For example, if you set nonce manually (different than what is on the block), sth might stop working. e.g. estimated fee doesn't work.

Reply:
Thanks for the clarification. What do you think about modifying the `get_nonce` function to make `block_id` an optional parameter? If it's not provided we can default to use `pending`. I suppose in the majority of cases we want to get nonce for the pending block. This way calling `get_nonce()` instead of `get_nonce('pending')` would be more convenient.

Reply:
In Rust optional works a little bit differently, if I refactor it to use it will be `get_nonce(Opion::None) | get_nonce(Option::Some('latest')).

Reply:
I see, if finding some kind of a workaround requires too much effort or is not possible, let's leave it as it is.